objects/src/accounts/code.rs

```diff
@@ -43,23 +40,10 @@ impl AccountCode {
     /// Returns an error if:
     /// - Compilation of the provided module fails.
     /// - The number of procedures exported from the provided module is greater than 256.
-    pub fn new(
-        account_id: AccountId,
-        account_module: ModuleAst,
-        assembler: &Assembler,
-    ) -> Result<Self, AccountError> {
-        // create a module with an account-specific path
-        // TODO: this should be eliminated as it results in a circular logic: we need an account ID
-        // to compile code, but we also need code root to build an account ID.
-        let module = Module::new(
-            LibraryPath::new(format!("{}_{}", Self::ACCOUNT_CODE_NAMESPACE_BASE, account_id))
-                .expect("valid path"),
-            account_module,
-        );
-
+    pub fn new(account_module: ModuleAst, assembler: &Assembler) -> Result<Self, AccountError> {
         // compile the module and make sure the number of exported procedures is within the limit
         let mut procedure_digests = assembler
-            .compile_module(&module, &mut AssemblyContext::for_module(false))
+            .compile_module(&account_module, None, &mut AssemblyContext::for_module(false))
             .map_err(AccountError::AccountCodeAssemblerError)?;
```

Comment:
In https://github.com/0xPolygonMiden/miden-vm/issues/1071 we had discussed that `compile_module` would return a mapping of `BTreeMap<Digest, ProcedureName>`.  Reviewing https://github.com/0xPolygonMiden/miden-vm/pull/1079 it looks like that approach was not taken?  Is there a reason why not?

I think it would be useful if the client could get access to this mapping information from the `AccountCode` object.  This could be done by changing the constructor return type to be `-> Result<(Self, BTreeMap<..>), AccountError>`.  However, we would need the `compile_module` method in the assembler to return the map.

Reply:
> In [0xPolygonMiden/miden-vm#1071](https://github.com/0xPolygonMiden/miden-vm/issues/1071) we had discussed that `compile_module` would return a mapping of `BTreeMap<Digest, ProcedureName>`. Reviewing [0xPolygonMiden/miden-vm#1079](https://github.com/0xPolygonMiden/miden-vm/pull/1079) it looks like that approach was not taken? Is there a reason why not?

One issue is that eventually we'll probably want to have the capability to compile a module with unnamed procedures. This would be useful for when we'd want to put account code on chain. In such situations, we'll probably need to strip all names off exported (and internal) procedures and rely purely on their indexes.

I guess we could change the return type to something like `BTreeMap<Digest, Option<ProcedureName>>`, but i'm not sure if that's as useful, and this probably requires more discussion.

Reply:
I see.  I guess the main point I'm trying to make here is if I'm a user and I define some account module with a bunch of procedures. I need a way to map those procedures to their corresponding mast root so I know which mast root is associated with which procedure.  If I don't know this mapping then I will not know how to invoke the correct account procedures from note scripts.